### PR TITLE
narrate number of results obtained on searching in PM UI

### PR DESCRIPTION
**Pull request by rohit21agrawal**
Fixes internal issue 395888
Related github issue: https://github.com/NuGet/Home/issues/5296

CC: @newrad0603

### CODE REVIEW

File: src/NuGet.Clients/NuGet.PackageManagement.UI/NuGetPackageManagerControlSearchTask.cs [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+        }
+        public void Start()
+        {
+            SetStatus(VsSearchTaskStatus.Started);
+            NuGetUIThreadHelper.JoinableTaskFactory.RunAsync(async () =>
+            {
+                await NuGetUIThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
+                _packageManagerControl.SearchPackagesAndRefreshUpdateCount(_searchQuery.SearchString, true, _searchCallback, this);
+            });
+            SetStatus(VsSearchTaskStatus.Completed);
```
Comment (Newrad0603) [inline src/NuGet.Clients/NuGet.PackageManagement.UI/NuGetPackageManagerControlSearchTask.cs]: I can't remember if RunAsync blocks the rest of the method execution, so if it doesn't you might want to move this into the RunAsync so the search isn't prematurely marked as complete.

Comment (rohit21agrawal) [reply src/NuGet.Clients/NuGet.PackageManagement.UI/NuGetPackageManagerControlSearchTask.cs]: @Newrad0603 thanks, fixed. RunAsync indeed does not block

**Review by emgarten — APPROVED** [review]
LGTM

File: src/NuGet.Clients/NuGet.PackageManagement.UI/NuGetPackageManagerControlSearchTask.cs [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+            Status = (uint)taskStatus;
+        }
+
+        private enum VsSearchTaskStatus : uint
+        {
+            Completed = 2,
+            Created = 0,
+            Error = 4,
+            Started = 1,
+            Stopped = 3
```
Comment (jainaashish) [inline src/NuGet.Clients/NuGet.PackageManagement.UI/NuGetPackageManagerControlSearchTask.cs]: nit: order these values

**Review by jainaashish — APPROVED** [review]
LGTM

File: src/NuGet.Clients/NuGet.PackageManagement.UI/NuGetPackageManagerControlSearchTask.cs [hunk 3]
```diff
@@ -0,0 +1,73 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Microsoft.VisualStudio.Shell.Interop;
+using NuGet.VisualStudio;
+
+namespace NuGet.PackageManagement.UI
+{
+    public sealed class NuGetPackageManagerControlSearchTask : IVsSearchTask
```
Comment (mishra14) [inline src/NuGet.Clients/NuGet.PackageManagement.UI/NuGetPackageManagerControlSearchTask.cs]: Please add comments for public classes and methods

**Review by mishra14 — APPROVED** [review]

**Pull request merged**